app/src/test/java/tttTDD/Java/GameTest.java

```diff
@@ -0,0 +1,169 @@
+package tttTDD.Java;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import tttTDD.Java.Interfaces.Player;
+
+public class GameTest {
+    Game game;
+    GameConfiguration gameConfig;
+
+    @Before
+    public void setup() {
+        gameConfig = new GameConfiguration(3, true);
+    }
+
+    @Test
+    public void PlayerShouldUpdateBoardWithinTheGame3x3() {
+        // Arrange
+        int playerInput = 3;
+        game = new Game(gameConfig);
+
+        // Act
+        game.performMove(playerInput);
+
+        // Assert
+        Assert.assertEquals(game.board.getSpace(playerInput), game.player1.getMarker());
+    }
+
+    @Test
+    public void GameShouldSwapPlayersAfterSuccessfulMoveOnTheBoard() {
+        int playerInput = 3;
+        game = new Game(gameConfig);
+
+        // Act
+        Player playerMakingMove = game.currentPlayer;
+        game.performMove(playerInput);
+
+        // Assert
+        Assert.assertNotEquals(playerMakingMove.getMarker(), game.currentPlayer.getMarker());
+    }
+
+    @Test
+    public void GameShouldGenerateComputerPlayerAsSecondPlayerIfConfigFlagIsTrue(){
+        game = new Game(gameConfig);
+
+        Assert.assertTrue(game.player2.isComputer());
+    }
+
+    @Test
+    public void ComputerShouldTakeTheFirstAvailableMoveAtLocation0() {
+        int playerInput = 3;
+        game = new Game(gameConfig);
+
+        game.performMove(playerInput);
+        game.performMove(2);
+
+        Assert.assertEquals(game.board.getSpace(0), game.player2.getMarker());
+    }
+
+    @Test
+    public void ComputerShouldTakeTheFirstAvailableMoveAtLocation4() {
+        int playerInput1 = 3;
+        int playerInput2 = 2;
+        int playerInput3 = 6;
+        game = new Game(gameConfig);
+
+        game.performMove(playerInput1);
+        game.performMove(2);
+        game.performMove(playerInput2);
+        game.performMove(7);
+        game.performMove(playerInput3);
+        game.performMove(9);
+
+        Assert.assertEquals(game.board.getSpace(4), game.player2.getMarker());
```

Comment:
The flow of this test is very hard to parse. I have to read the implementation of `performMove`, `getFirstAvailableMove`, and `getAvailableMoves` in order to understand _when_ the computer takes _what_ specific moves. And even at that point, I have to keep a mental tally of those moves line by line, peaking up at the definitions of `playerInputX` and remembering to ignore the values passed on every other call.

Reply:
A lot of this comes down to the design of the method under test, but even without changing that, you could make this clearer by adding comments to each step and/or ditching the playerInput variables.

Reply:
I do agree with you that it can be difficult to follow along. This was the result of testing the game conclusion without a game loop or direct player input from an external source like a `Scanner`

Reply:
Yeah, I can see how that would help for sure. Not something to stress at the moment, but being able to seed a current state of a board to test an individual move could be a good strategy to learn when dealing with complex series of events needed to get to a specific point in an app to test it. -- Something we can follow back upon at a later point, plenty to digest now as it is.